Refactor headings to follow Google Developer Style Guide

Srishti-j18 opened the pull request:
Fixed #12466

~For now, I am marking this as a draft. It - still needs some work.~
- [x] I have a few queries. I am commenting all of them.
- [x] changed h1 (main) headers of doc files according to the google developer style guild .

lb- commented:
@Srishti-j18 for now, can you scope this PR to only updating the casing (capitalisation) of the page h1 titles. Thanks for digging into this though.

>     1. As you mentioned in [this documentation](https://docs.wagtail.org/en/latest/advanced_topics/images/feature_detection.html) as well as others regarding Feature detection—should these changes apply only to headings, or should they also appear in the text? I have applied them to both in my commit.

The word feature or detection should not be capitalised, unless they are at the start of a sentence or title. 

Sorry if there was confusion here, these words are not proper nouns. 

>     2. According to the [Google Developer Guide](https://developers.google.com/style/headings) on headings and title text, I updated a few headings. Could you confirm if headings like [ordering](https://github.com/wagtail/wagtail/blob/987bd2a05463856a692ab995505bc2c29d059e2a/docs/advanced_topics/api/v2/usage.md?plain=1#L169) or [filtering](https://github.com/wagtail/wagtail/blob/987bd2a05463856a692ab995505bc2c29d059e2a/docs/advanced_topics/api/v2/usage.md?plain=1#L263) should also change to Order and Filter respectively? Should we update related text consistently throughout as well?

Good find, but for now please leave these as is, this might be a good candidate for a new issue/PR later. Let's just solve one thing at a time. 

>     3. Based on the previous point, should [Managing the Reference Index](https://docs.wagtail.org/en/latest/advanced_topics/reference_index.html) also change to Manage the Reference Index/ Manage the reference index? Should it be capitalized exactly like this? for this change we also have to change the link of this managing_the_reference_index->manage_the_reference_index ig.

Good question, let's change this one only to `Mange the reference index`, I think lowe case makes sense. We don't use this as a proper noun throughout the rest of that doc so I think the title should be lower case.

Leave the URL (filename) as is, these do not have to be 100% aligned. The URL serves a different purpose to the page title. 

>     4. As per [Google's guide](https://developers.google.com/style/headings), no heading should be empty. However, I noticed that a few headings, such as [this one](https://github.com/wagtail/wagtail/blob/987bd2a05463856a692ab995505bc2c29d059e2a/docs/advanced_topics/api/v2/usage.md?plain=1#L561) and [this one](https://github.com/wagtail/wagtail/blob/987bd2a05463856a692ab995505bc2c29d059e2a/docs/advanced_topics/api/v2/usage.md?plain=1#L627), are empty. Should we add content ?

Good find, for now, leave this as is. If you have some suggestions for what useful content to put in these empty headings. Please prepare a separate PR. 

> I have only viewed the URLs you mentioned in the issues. Once you confirm all these points, it would be better to make changes to the other URLs as well.

Please do a review, but again a small PR is a good PR. Let's focus only on the h1 (main) headers for now. Then, start digging into other heading levels.

The main goal here for now is to ensure our page titles (h1)s align nicely and are consistent in the table of contents.

lb- commented on docs/advanced_topics/images/feature_detection.md:
```diff
@@ -1,14 +1,14 @@
 ## Installation
 
-Two third-party tools are known to work with Wagtail: One based on [OpenCV](https://opencv.org/) for general feature detection and one based on [Rustface](https://github.com/torchbox/rustface-py/) for face detection.
+Two third-party tools are known to work with Wagtail: One based on [OpenCV](https://opencv.org/) for general Feature detection and one based on [Rustface](https://github.com/torchbox/rustface-py/) for face detection.
```
```suggestion
Two third-party tools are known to work with Wagtail: One based on [OpenCV](https://opencv.org/) for general feature detection and one based on [Rustface](https://github.com/torchbox/rustface-py/) for face detection.
```

Srishti-j18 commented:
> @Srishti-j18 for now, can you scope this PR to only updating the casing (capitalisation) of the page h1 titles. Thanks for digging into this though.
> 
> > ```
> > 1. As you mentioned in [this documentation](https://docs.wagtail.org/en/latest/advanced_topics/images/feature_detection.html) as well as others regarding Feature detection—should these changes apply only to headings, or should they also appear in the text? I have applied them to both in my commit.
> > ```
> 
> The word feature or detection should not be capitalised, unless they are at the start of a sentence or title.
> 
> Sorry if there was confusion here, these words are not proper nouns.
> 
> > ```
> > 2. According to the [Google Developer Guide](https://developers.google.com/style/headings) on headings and title text, I updated a few headings. Could you confirm if headings like [ordering](https://github.com/wagtail/wagtail/blob/987bd2a05463856a692ab995505bc2c29d059e2a/docs/advanced_topics/api/v2/usage.md?plain=1#L169) or [filtering](https://github.com/wagtail/wagtail/blob/987bd2a05463856a692ab995505bc2c29d059e2a/docs/advanced_topics/api/v2/usage.md?plain=1#L263) should also change to Order and Filter respectively? Should we update related text consistently throughout as well?
> > ```
> 
> Good find, but for now please leave these as is, this might be a good candidate for a new issue/PR later. Let's just solve one thing at a time.
> 
> > ```
> > 3. Based on the previous point, should [Managing the Reference Index](https://docs.wagtail.org/en/latest/advanced_topics/reference_index.html) also change to Manage the Reference Index/ Manage the reference index? Should it be capitalized exactly like this? for this change we also have to change the link of this managing_the_reference_index->manage_the_reference_index ig.
> > ```
> 
> Good question, let's change this one only to `Mange the reference index`, I think lowe case makes sense. We don't use this as a proper noun throughout the rest of that doc so I think the title should be lower case.
> 
> Leave the URL (filename) as is, these do not have to be 100% aligned. The URL serves a different purpose to the page title.
> 
> > ```
> > 4. As per [Google's guide](https://developers.google.com/style/headings), no heading should be empty. However, I noticed that a few headings, such as [this one](https://github.com/wagtail/wagtail/blob/987bd2a05463856a692ab995505bc2c29d059e2a/docs/advanced_topics/api/v2/usage.md?plain=1#L561) and [this one](https://github.com/wagtail/wagtail/blob/987bd2a05463856a692ab995505bc2c29d059e2a/docs/advanced_topics/api/v2/usage.md?plain=1#L627), are empty. Should we add content ?
> > ```
> 
> Good find, for now, leave this as is. If you have some suggestions for what useful content to put in these empty headings. Please prepare a separate PR.
> 
> > I have only viewed the URLs you mentioned in the issues. Once you confirm all these points, it would be better to make changes to the other URLs as well.
> 
> Please do a review, but again a small PR is a good PR. Let's focus only on the h1 (main) headers for now. Then, start digging into other heading levels.
> 
> The main goal here for now is to ensure our page titles (h1)s align nicely and are consistent in the table of contents.

ok , I have viewed all the urls and changed only on the h1 (main) headers for now.

Srishti-j18 commented:
https://github.com/wagtail/wagtail/pull/12471#issuecomment-2438910421

And one more thing: Should I create a separate issue for each of the points I’ve commented on and resolve them one by one?

lb- commented on docs/deployment/under_the_hood.md:
```diff
@@ -1,4 +1,4 @@
-# Wagtail Deployment: Under the Hood
+# Wagtail Deployment: under the Hood
```
```suggestion
# Deployment: Under the hood
```

This title is not ideal but for now let's just fix the capitalisation and remove the unnecessary Wagtail name.

The word after the colon should probably be capitalised.

lb- reviewed (approved):
@Srishti-j18  thank you for this and for following through on the feedback.

I have one small comment but will fix this up when merging.

I will push to your branch shortly.

lb- commented:
> And one more thing: Should I create a separate issue for each of the points I’ve commented on and resolve them one by one?

@Srishti-j18 I think we should be careful not to overdo it with the fixes.

The long term goal is to align better with the linked Google Developer Documentation Style Guide but I am not super sure on how much code / docs churn we are willing to do here.

What I might do is chat with the core team next week and review whether we should create a new parent issue for wider adoption.

In the mean time, feel free to do a small scale PR that makes a few improvements to titles in one section of the docs. e.g. `advanced_topics/customisation` (while at it, this could probably be updated at the URL level to use `customization` (US spelling).

FYI @activus-d  / @thibaudcolas  - it appears that we documented the adoption of the new style guide but did not create a new issue to update our docs progressively, not sure if that was intentional. https://github.com/wagtail/wagtail/pull/10634 & https://github.com/wagtail/wagtail/issues/10608
